data-migration: add doc for migrate from csv files and from aurora (#7147)
pingcap/docs: data-migration/migrate-flat-file-tidb.md

- [Get the target database privileges required for TiDB Lightning](/tidb-lightning/tidb-lightning-faq.md#what-are-the-privilege-requirements-for-the-target-database).

## Step 1. Prepare the CSV files

Put all the CSV files in the same directory. If you need TiDB Lightning to recognize all CSV files, the file name should satisfy the following requirements:

- If a CSV file contains the data for an entire table, name the file `${db_name}.${table_name}.csv`.
- If the data of one table is separated into multiple CSV files, append a numeric suffix to these CSV files. For example, `${db_name}.${table_name}.003.csv`. The numeric suffixes may not be consecutive, but must be in ascending order. You might need to add extra zeros before the number to ensure all the suffixes are in the same length.

## Step 2. Create the target table schema

Because CSV files do not contain schema information, before importing data into TiDB, you need to create the target table schema. You can create the target table schema by either of the following two methods:

* **Method 1**: create the target table schema using TiDB Lightning.

    1. Write SQL files that contain the required DDL statements.

>         - Put `CREATE DATABASE` statements in the `${db_name}-schema-create.sql` files.
>         - Put `CREATE TABLE` statements in the `${db_name}.${table_name}-schema.sql` files.

Author: ```suggestion
        - Add `CREATE DATABASE` statements in the `${db_name}-schema-create.sql` files.
        - Add `CREATE TABLE` statements in the `${db_name}.${table_name}-schema.sql` files.
```

    2. During the migration, add the following configuration in `tidb-lightning.toml`:

        ```toml
        [mydumper]

>         no-schema = false # To create database and table using Lightning, set the value to false

Reviewer: 这里是不是 schema and table 呢？
```suggestion
        no-schema = false # To create schema and table using Lightning, set the value to false
```

Author: 这里我感觉是跟上一步里的 `CREATE DATABASE` 和 `CREATE TABLE` 对应的，所以写了 `create database and table`

Author: 又想了下，还是 target table schema 更合适。谢谢~

not-null = false
# If `not-null` is set to false (CSV contains NULL),
# The following value is parsed as NULL.
null = '\N'
# Whether to parse the backslash as an escape character.
backslash-escape = true
# Whether to trim the line that ends with a separator.
trim-last-separator = false

# The target cluster.
host = "${ip}"
port = ${port}
user = "${user_name}"
password = "${password}"
# The table schema is obtained from the TiDB status port.
status-port = ${port} # e.g.: 10080
# Address of the PD cluster
pd-addr = "${ip}:${port}" # e.g.: 172.16.31.3:2379. When backend = "local", you must specify status-port and pd-addr. Otherwise, the import will be abnormal.
```

For more information on the configuration file, refer to [TiDB Lightning Configuration](/tidb-lightning/tidb-lightning-configuration.md).

## Step 4. Speed up the import (optional)

When you import data from CSV files with a uniform size of about 256 MiB, TiDB Lightning works in the best performance. However, if you import data from a single large CSV file, TiDB Lightning can only use one thread to process the import by default, which might slow down the import speed.

> To speed up the import, you can split a large CSV file into smaller ones. For CSV files in common formats, it is hard to quickly locate the beginning and ending positions of each line. Therefore,  TiDB Lightning does not automatically split CSV files by default. But if your CSV files to be imported meet certain format requirements, you can enable the `strict-format` mode. In this mode, TiDB Lightning automatically splits CSV files into multiple files, each in about 256 MiB, and processes them in parallel.

Reviewer 2: ```suggestion
To speed up the import, you can split a large CSV file into smaller ones. For a CSV file in a common format, before TiDB Lightning reads the entire file, it is hard to quickly locate the beginning and ending positions of each line. Therefore,  TiDB Lightning does not automatically split CSV files by default. But if your CSV files to be imported meet certain format requirements, you can enable the `strict-format` mode. In this mode, TiDB Lightning automatically splits a single large CSV file into multiple files, each in about 256 MiB, and processes them in parallel.
```

Reviewer 2: 补充了中文中的 “在没有读取整个文件的情况下”的翻译

> **Note:**
>

> > If the CSV file is not in strict format but the `strict-format` mode is set to `true` by mistake, a field that spans multiple lines will be split into two fields. This causes the parsing to fail, and TiDB Lightning might import the corrupted data without reporting an error.

Reviewer 2: ```suggestion
> If a CSV file is not in a strict format but the `strict-format` mode is set to `true` by mistake, a field that spans multiple lines will be split into two fields. This causes the parsing to fail, and TiDB Lightning might import the corrupted data without reporting any error.
```

* **Method 1**: create the target table schema using TiDB Lightning.

    1. Write SQL files that contain the required DDL statements.

        - Add `CREATE DATABASE` statements in the `${db_name}-schema-create.sql` files.
        - Add `CREATE TABLE` statements in the `${db_name}.${table_name}-schema.sql` files.

    2. During the migration, add the following configuration in `tidb-lightning.toml`:

        ```toml
        [mydumper]
        no-schema = false # To create a target table schema using Lightning, set the value to false.
        ```

* **Method 2**: create the target table schema manually.

    During the migration, add the following configuration in `tidb-lightning.toml`:

    ```toml
    [mydumper]
    no-schema = true # If you have already created the target table schema, set the value to true, which means skipping the schema creation.
    ```

## Step 3. Create the configuration file

Create a `tidb-lightning.toml` file with the following content:

{{< copyable "shell-regular" >}}

```toml
[lightning]
# Log
level = "info"
file = "tidb-lightning.log"

[tikv-importer]

> # "local": Default backend. The local backend is used to import large volumes of data (1 TiB or more). During the import, the target TiDB cluster cannot provide any service.
> # "tidb": The "tidb" backend is used to import data less than 1 TiB. During the import, the target TiDB cluster can provide service normally.

Reviewer 2: ```suggestion
# "local": Default backend. The local backend is recommended to import large volumes of data (1 TiB or more). During the import, the target TiDB cluster cannot provide any service.
# "tidb": The "tidb" backend is recommended to import data less than 1 TiB. During the import, the target TiDB cluster can provide service normally.
```
